File: Anul_2/MPP/ChatRMI/ChatPersistence/src/main/java/chat/persistence/ICrudRepository.java
package chat.persistence;

import chat.model.Identifiable;

public interface ICrudRepository<ID, E extends Identifiable<ID>> {
    void save(E e);
    void delete(ID id);
    E findOne(ID id);
    void update(ID id, E e);
    Iterable<E> getAll();
}

File: Anul_3/Semestrul 1/MAP - Restanta/MAP_CSharp/MAP_CSharp/MAP_CSharp/repositories/ICrudRepository.cs
using System;
using MAP_CSharp.model;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace MAP_CSharp.repositories
{
    public interface ICrudRepository<ID, E> where E : Entitate<ID>
    {
        int Size();
        E FindOne(ID id);
        IEnumerable<E> FindAll();
        E Save(E e);
        E Delete(ID id);
        E Update(E entity);


    }
}
